packages/mobile/src/verify/VerificationFailedModal.tsx

```diff
@@ -44,56 +44,88 @@ export function VerificationFailedModal(props: Props) {
 
   const userBalanceInsufficient =
     props.verificationStatus === VerificationStatus.InsufficientBalance
+  const saltQuotaExceeded = props.verificationStatus === VerificationStatus.SaltQuotaExceeded
 
   const isVisible =
     (props.verificationStatus === VerificationStatus.Failed ||
       props.verificationStatus === VerificationStatus.RevealAttemptFailed ||
-      userBalanceInsufficient) &&
+      userBalanceInsufficient ||
+      saltQuotaExceeded) &&
     !isDismissed
 
   const allowEnterCodes = props.verificationStatus === VerificationStatus.RevealAttemptFailed
   // Only prompt forno switch if not already in forno mode and failure
   // wasn't due to insuffuicient balance
   const promptRetryWithForno =
-    props.retryWithForno &&
-    !props.fornoMode &&
-    props.verificationStatus !== VerificationStatus.InsufficientBalance
+    props.retryWithForno && !props.fornoMode && !userBalanceInsufficient && !saltQuotaExceeded
 
-  return promptRetryWithForno ? (
+  if (promptRetryWithForno) {
     // Retry verification with forno with option to skip verificaion
-    <WarningModal
-      isVisible={isVisible}
-      header={t('retryWithFornoModal.header')}
-      body1={t('retryWithFornoModal.body1')}
-      body2={t('retryWithFornoModal.body2')}
-      continueTitle={t('retryWithFornoModal.retryButton')}
-      cancelTitle={t('education.skip')}
-      onCancel={onSkip}
-      onContinue={onRetry}
-    />
-  ) : allowEnterCodes ? (
+    return (
+      <WarningModal
+        isVisible={isVisible}
+        header={t('retryWithFornoModal.header')}
+        body1={t('retryWithFornoModal.body1')}
+        body2={t('retryWithFornoModal.body2')}
+        continueTitle={t('retryWithFornoModal.retryButton')}
+        cancelTitle={t('education.skip')}
+        onCancel={onSkip}
+        onContinue={onRetry}
+      />
+    )
+  }
+
+  if (allowEnterCodes) {
     // Option to enter codes if reveal attempt failed
+    return (
+      <WarningModal
+        isVisible={isVisible}
+        header={t('failModal.header')}
+        body1={t('failModal.body1')}
+        body2={t('failModal.enterCodesBody')}
+        continueTitle={t('education.skip')}
+        cancelTitle={t('global:goBack')}
+        onCancel={onDismiss}
+        onContinue={onSkip}
```

Comment:
Doesn't `onSkip` cancel verification? Will this take users to the the code screen somehow?

Reply:
`onSkip` will move user to Home screen and `onDismiss` would dismiss the modal and they would see the Enter Codes screes (I've change "Go Back" to "Enter Codes" for dismiss action as well)

Reply:
Makes sense, I misunderstood what `onDismiss` does in this case